### PR TITLE
TF-801 Enable my vacation responder

### PR DESCRIPTION
**Pull request by dab246**
#801 

### Resolved

https://user-images.githubusercontent.com/80730648/185177539-4112d0e5-df1b-4285-be80-dfd8e6d98a36.mov

### CODE REVIEW

**Comment by hoangdat** [comment]
- Menu item should be `Vacation` instead of `Configuration` :)
- How we can change the `accentColor` for Date/Time picker?
- Please change the toggle button more beautiful :)

File: lib/features/manage_account/presentation/configuration/vacation/vacation_view.dart [hunk 1]
```diff
@@ -430,25 +438,33 @@ class VacationView extends GetWidget<VacationController> {
...
-          color: Colors.black.withOpacity(opacity),
-          fontSize: 16))
-      ..textDecoration((VacationInputDecorationBuilder()
-          ..setContentPadding(const EdgeInsets.symmetric(
-              vertical: BuildUtils.isWeb ? 16 : 12,
-              horizontal: 12))
-          ..setHintStyle(TextStyle(
-              color: AppColor.colorHintInputCreateMailbox.withOpacity(opacity),
+    return Obx(() {
+      return (TextFieldBuilder()
```
Comment (hoangdat) [inline lib/features/manage_account/presentation/configuration/vacation/vacation_view.dart:442]: please note that, vacation also support `htmlBody`. Please do it the same with composer

Comment (hoangdat) [reply lib/features/manage_account/presentation/configuration/vacation/vacation_view.dart:442]: but at the moment, no problem with only support `textBody`

Comment (hoangdat) [reply lib/features/manage_account/presentation/configuration/vacation/vacation_view.dart:442]: we will support `htmlBody` in https://github.com/linagora/tmail-flutter/issues/858

Comment (dab246) [reply lib/features/manage_account/presentation/configuration/vacation/vacation_view.dart:442]: oke

**Comment by dab246** [comment]
> * Menu item should be `Vacation` instead of `Configuration` :)
> * How we can change the `accentColor` for Date/Time picker?
> * Please change the toggle button more beautiful :)

1. IMO, We can waiting new design
2. Yes, we can change it. what are you want `color`?
3. Oke. i update it

**Comment by dab246** [comment]
### Video demo updated

https://user-images.githubusercontent.com/80730648/185292609-a7f1a72e-85cd-458b-b007-76246734f45d.mov

File: lib/features/manage_account/presentation/model/account_menu_item.dart [hunk 2]
```diff
@@ -7,9 +7,7 @@ enum AccountMenuItem {
   profiles,
   languageAndRegion,
   emailRules,
-}
-
-extension AccountMenuItemExtension on AccountMenuItem {
+  configuration;
```
Comment (hoangdat) [inline lib/features/manage_account/presentation/model/account_menu_item.dart:10]: vacation

File: lib/features/manage_account/presentation/menu/manage_account_menu_controller.dart [hunk 3]
```diff
@@ -44,6 +41,7 @@ class ManageAccountMenuController extends BaseController {
         AccountMenuItem.profiles,
         AccountMenuItem.emailRules,
         AccountMenuItem.languageAndRegion,
+        AccountMenuItem.configuration,
```
Comment (hoangdat) [inline lib/features/manage_account/presentation/menu/manage_account_menu_controller.dart:44]: vacation

File: lib/features/manage_account/presentation/manage_account_dashboard_view.dart [hunk 4]
```diff
@@ -154,6 +187,8 @@ class ManageAccountDashBoardView extends GetWidget<ManageAccountDashBoardControl
           } else {
             return const SizedBox.shrink();
           }
+        case AccountMenuItem.configuration:
+          return ConfigurationView();
```
Comment (hoangdat) [inline lib/features/manage_account/presentation/manage_account_dashboard_view.dart:191]: `VacationView`

File: lib/features/manage_account/presentation/model/account_menu_item.dart (hunk 2 above)
Comment (dab246) [reply lib/features/manage_account/presentation/model/account_menu_item.dart]: oke

File: lib/features/manage_account/presentation/menu/manage_account_menu_controller.dart (hunk 3 above)
Comment (dab246) [reply lib/features/manage_account/presentation/menu/manage_account_menu_controller.dart]: oke

File: lib/features/manage_account/presentation/manage_account_dashboard_view.dart (hunk 4 above)
Comment (dab246) [reply lib/features/manage_account/presentation/manage_account_dashboard_view.dart]: oke

**Review by hoangdat — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

File: lib/features/mailbox_dashboard/presentation/mixin/user_setting_popup_menu_mixin.dart [hunk 5]
```diff
@@ -44,7 +44,7 @@ mixin UserSettingPopupMenuMixin {
   Widget _settingAction(BuildContext context, Function? onCallBack) {
     return PopupMenuItemWidget(
         _imagePaths.icSetting,
-        AppLocalizations.of(context).settings,
+        AppLocalizations.of(context).manage_account,
```
Comment (ManhNTX) [inline lib/features/mailbox_dashboard/presentation/mixin/user_setting_popup_menu_mixin.dart:47]: IMO! using camel case

**Review by ManhNTX — APPROVED** [review]

File: lib/features/mailbox_dashboard/presentation/mixin/user_setting_popup_menu_mixin.dart (hunk 5 above)
Comment (dab246) [reply lib/features/mailbox_dashboard/presentation/mixin/user_setting_popup_menu_mixin.dart:47]: ok

**Pull request merged**